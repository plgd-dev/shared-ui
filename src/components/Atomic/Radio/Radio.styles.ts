import { css } from '@emotion/react'

export const radio = css`
    fieldset {
        padding: 0;
        margin: 0 -12px;
        border: 0;
        display: flex;
    }
`

export const radioItem = css`
    margin: 0 12px;
`
