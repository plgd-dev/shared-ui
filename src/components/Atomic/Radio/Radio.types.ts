import { SyntheticEvent } from 'react'

export type RadioItem = {
    label?: string
    value: string
}

export type Props = {
    className?: string
    defaultValue?: string
    error?: boolean
    id?: string
    items: RadioItem[]
    name: string
    onChange: (value: string, e: SyntheticEvent) => void
}
