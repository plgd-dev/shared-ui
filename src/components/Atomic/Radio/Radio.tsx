import React, { FC, SyntheticEvent } from 'react'
import { Props } from './Radio.types'
import * as styles from './Radio.styles'
import Checkbox from '../Checkbox'
import { isFunction } from 'lodash'

const Radio: FC<Props> = (props) => {
    const { id, className, items, name, defaultValue, error, onChange } = props
    const handleChange = (value: string, e: SyntheticEvent) => {
        isFunction(onChange) && onChange(value, e)
    }

    return (
        <div className={className} css={styles.radio} id={id}>
            <fieldset>
                {items.map((item) => (
                    <Checkbox
                        css={styles.radioItem}
                        defaultChecked={defaultValue === item.value}
                        error={error}
                        id={item.value}
                        key={item.value}
                        label={item.label}
                        name={name}
                        onChange={(e) => handleChange(item.value, e)}
                        type='radio'
                    />
                ))}
            </fieldset>
        </div>
    )
}

Radio.displayName = 'Radio'

export default Radio
