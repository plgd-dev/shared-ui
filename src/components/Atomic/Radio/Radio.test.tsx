import React from 'react'
import { render } from '@testing-library/react'
import Radio from './Radio'

describe('<Radio>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <Radio items={[{ value: '1' }, { value: '2' }]} name='radio1' onChange={() => {}} />
                <Radio
                    items={[
                        { value: '1', label: '1' },
                        { value: '2', label: '2' },
                    ]}
                    onChange={() => {}}
                    name='radio1'
                />
                <Radio
                    defaultValue='2'
                    onChange={() => {}}
                    items={[
                        { value: '1', label: '1' },
                        { value: '2', label: '2' },
                    ]}
                    name='radio1'
                />
                <Radio
                    error={true}
                    onChange={() => {}}
                    items={[
                        { value: '1', label: '1' },
                        { value: '2', label: '2' },
                    ]}
                    name='radio1'
                />
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
