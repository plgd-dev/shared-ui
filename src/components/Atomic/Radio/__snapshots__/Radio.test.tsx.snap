// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 fieldset {
  padding: 0;
  margin: 0 -12px;
  border: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  position: relative;
  overflow: hidden;
  padding-bottom: 2px;
  margin: 0 12px;
}

.emotion-2 {
  width: 1px;
  height: 1px;
  position: absolute;
  left: -10px;
}

.emotion-2:indeterminate~.emotion-4:before {
  opacity: 1;
}

.emotion-2:checked~.emotion-4:after {
  opacity: 1;
}

.emotion-3 {
  width: 16px;
  height: 16px;
  -webkit-flex: 0 0 16px;
  -ms-flex: 0 0 16px;
  flex: 0 0 16px;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  position: relative;
  box-sizing: border-box;
  top: 2px;
}

.emotion-3:before,
.emotion-3:after {
  content: '';
  display: block;
  -webkit-transition: opacity 0.25s;
  transition: opacity 0.25s;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  width: 10px;
  border-radius: 2px;
  opacity: 0;
}

.emotion-3:before {
  height: 2px;
}

.emotion-3:after {
  height: 10px;
}

.emotion-14 {
  padding-left: 8px;
  cursor: pointer;
}

.emotion-34 {
  width: 16px;
  height: 16px;
  -webkit-flex: 0 0 16px;
  -ms-flex: 0 0 16px;
  flex: 0 0 16px;
  border: 1px solid;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  position: relative;
  box-sizing: border-box;
  top: 2px;
}

.emotion-34:before,
.emotion-34:after {
  content: '';
  display: block;
  -webkit-transition: opacity 0.25s;
  transition: opacity 0.25s;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  width: 10px;
  border-radius: 2px;
  opacity: 0;
}

.emotion-34:before {
  height: 2px;
}

.emotion-34:after {
  height: 10px;
}

<div>
    <div
      class="emotion-0"
    >
      <fieldset>
        <label
          class="emotion-1"
          id="1"
        >
          <input
            class="emotion-2"
            name="radio1"
            type="radio"
          />
          <span
            class="emotion-3 emotion-4"
          />
        </label>
        <label
          class="emotion-1"
          id="2"
        >
          <input
            class="emotion-2"
            name="radio1"
            type="radio"
          />
          <span
            class="emotion-3 emotion-4"
          />
        </label>
      </fieldset>
    </div>
    <div
      class="emotion-0"
    >
      <fieldset>
        <label
          class="emotion-1"
          id="1"
        >
          <input
            class="emotion-2"
            name="radio1"
            type="radio"
          />
          <span
            class="emotion-3 emotion-4"
          />
          <div
            class="emotion-14"
          >
            1
          </div>
        </label>
        <label
          class="emotion-1"
          id="2"
        >
          <input
            class="emotion-2"
            name="radio1"
            type="radio"
          />
          <span
            class="emotion-3 emotion-4"
          />
          <div
            class="emotion-14"
          >
            2
          </div>
        </label>
      </fieldset>
    </div>
    <div
      class="emotion-0"
    >
      <fieldset>
        <label
          class="emotion-1"
          id="1"
        >
          <input
            class="emotion-2"
            name="radio1"
            type="radio"
          />
          <span
            class="emotion-3 emotion-4"
          />
          <div
            class="emotion-14"
          >
            1
          </div>
        </label>
        <label
          class="emotion-1"
          id="2"
        >
          <input
            checked=""
            class="emotion-2"
            name="radio1"
            type="radio"
          />
          <span
            class="emotion-3 emotion-4"
          />
          <div
            class="emotion-14"
          >
            2
          </div>
        </label>
      </fieldset>
    </div>
    <div
      class="emotion-0"
    >
      <fieldset>
        <label
          class="emotion-1"
          id="1"
        >
          <input
            class="emotion-2"
            name="radio1"
            type="radio"
          />
          <span
            class="emotion-34 emotion-4"
          />
          <div
            class="emotion-14"
          >
            1
          </div>
        </label>
        <label
          class="emotion-1"
          id="2"
        >
          <input
            class="emotion-2"
            name="radio1"
            type="radio"
          />
          <span
            class="emotion-34 emotion-4"
          />
          <div
            class="emotion-14"
          >
            2
          </div>
        </label>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
